### PR TITLE
[JENKINS-73695] Prevent unnecessary horizontal scrollbar in Firefox

### PR DESCRIPTION
**Pull request by scherler**
See [JENKINS-73695](https://issues.jenkins.io/browse/JENKINS-73695).

### Testing done

Interactive tests with Firefox

### Proposed changelog entries

- Prevent unnecessary horizontal scrollbar in Firefox.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Comment by scherler** [comment]
Follow up for https://github.com/jenkinsci/jenkins/pull/9667 which produced a horizontal scrollbar in FF

**Comment by janfaracik** [comment]
You could remove the top margin from `.jenkins-icon-size`, e.g.

```scss
.jenkins-icon-size {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin: 0 0 var(--section-padding) 0;

  ...
}
```

**Comment by scherler** [comment]
@janfaracik just tried that but it did not work in FF

**Comment by janfaracik** [comment]
Seems like it's the `::after` pseudo element on the table causing the overflow in Firefox, maybe have a media query to only apply that styling for non Firefox browsers?

**Comment by scherler** [comment]
@janfaracik added the media query now instead of reducing the width

File: war/src/main/scss/components/_table.scss [hunk 1]
```diff
@@ -13,13 +13,15 @@
...
   margin-bottom: var(--section-padding);
 
-  &::before {
-    content: "";
-    position: absolute;
-    inset: -4px -4px -2px;
-    border: var(--card-border-width) solid var(--table-border-color);
-    border-radius: inherit;
-    pointer-events: none;
+  @supports (not (-moz-appearance: none)) {
```
Comment (janfaracik) [inline war/src/main/scss/components/_table.scss:16]:
```suggestion
  // The '::before' pseudo-element is causing overflow issues in Firefox, so disable it for that browser
  @supports (not (-moz-appearance: none)) {
```

LGTM, would be worth adding a comment above the media query explaining why we have it.

**Review by janfaracik — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge

**Comment by daniel-beck** [comment]
This should have a different Jira issue than the original issue so this can be backported into LTS.

**Pull request merged**